crates/bin/pcli/src/command/tx/auction/dutch.rs

```diff
@@ -366,6 +370,85 @@ impl DutchCmd {
     }
 }
 
+async fn all_dutch_auction_states(
+    view_client: &mut impl ViewClient,
+    source: impl Into<AddressIndex>,
+) -> Result<Vec<(AuctionId, DutchAuction, u64)>> {
+    fetch_dutch_auction_states(view_client, source, true).await
+}
+
+async fn active_dutch_auction_states(
+    view_client: &mut impl ViewClient,
+    source: impl Into<AddressIndex>,
+) -> Result<Vec<(AuctionId, DutchAuction, u64)>> {
+    fetch_dutch_auction_states(view_client, source, false).await
+}
+
+async fn fetch_dutch_auction_states(
+    view_client: &mut impl ViewClient,
+    source: impl Into<AddressIndex>,
+    include_inactive: bool,
+) -> Result<Vec<(AuctionId, DutchAuction, u64)>> {
+    let auctions = view_client
+        .auctions(Some(source.into()), include_inactive, true)
+        .await?
+        .into_iter()
+        .filter_map(|(id, _, local_seq, state, _)| {
+            if let Some(state) = state {
+                if let Ok(da) = DutchAuction::decode(state.value) {
+                    Some((id, da, local_seq))
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect();
+    Ok(auctions)
+}
+/// Return all the auctions that need to be ended, based on our local view of the chain state.
+async fn auctions_to_end(view_client: &mut impl ViewClient, source: u32) -> Result<Vec<AuctionId>> {
+    let auctions = active_dutch_auction_states(view_client, source).await?;
+
+    let auction_ids = auctions
```

Comment:
non-blocking nit: this doesn't need a `let` binding, since we immediately return it.

ditto for `auctions_to_withdraw`, below

Reply:
that's subjective, but i prefer an extra binding to make it easy to skim the code and know what it returns